src/irr/asset/COBJMeshFileLoader.cpp

```diff
@@ -355,9 +335,93 @@ asset::SAssetBundle COBJMeshFileLoader::loadAsset(io::IReadFile* _file, const as
 		// eat up rest of line
 		bufPtr = goNextLine(bufPtr, bufEnd);
 	}	// end while(bufPtr && (bufPtr-buf<filesize))
-	// Clean up the allocate obj _file contents
-	delete [] buf;
 
+    core::vector<core::vectorSIMDf> newNormals;
+    auto doRecalcNormals = [&vertices,&newNormals](const core::vector<uint32_t>& _ixs) {
+        memset(newNormals.data(), 0, sizeof(core::vectorSIMDf)*newNormals.size());
+
+        auto minmax = std::minmax_element(_ixs.begin(), _ixs.end());
+        const uint32_t maxsz = *minmax.second - *minmax.first;
+        const uint32_t min = *minmax.first;
+        
+        newNormals.resize(maxsz, core::vectorSIMDf(0.f));
+        for (size_t i = 0ull; i < _ixs.size(); i += 3ull)
+        {
+            core::vectorSIMDf v1, v2, v3;
+            v1.set(vertices[_ixs[i-min+0u]].pos);
+            v2.set(vertices[_ixs[i-min+1u]].pos);
+            v3.set(vertices[_ixs[i-min+2u]].pos);
+            v1.makeSafe3D();
+            v2.makeSafe3D();
+            v3.makeSafe3D();
+            core::vectorSIMDf normal(core::plane3dSIMDf(v1, v2, v3).getNormal());
+            newNormals[_ixs[i-min+0u]] += normal;
+            newNormals[_ixs[i-min+1u]] += normal;
+            newNormals[_ixs[i-min+2u]] += normal;
+        }
+        for (uint32_t ix : _ixs)
+            vertices[ix].normal32bit = asset::quantizeNormal2_10_10_10(newNormals[ix-min]);
+    };
+
+    {
+        uint64_t ixBufOffset = 0ull;
+        for (size_t i = 0ull; i < submeshes.size(); ++i)
+        {
+            if (submeshWasLoadedFromCache[i])
+                continue;
+            if (recalcNormals[i])
+                doRecalcNormals(indices[i]);
+
+            submeshes[i]->setIndexCount(indices[i].size());
+            submeshes[i]->setIndexType(EIT_32BIT);
+            submeshes[i]->getIndexBufferBinding()->offset = ixBufOffset;
+            ixBufOffset += indices[i].size()*4ull;
+
+            const bool hasUV = !std::isnan(vertices[indices[i][0]]);
```

Comment:
`std::isnan` is broken on GCC `-ffast-math` we should replace all usages of `std::isnan`, `std::nan`, and `isinf`, etc. with our own `core::` implementations that do the bitmask checking magic themselves

(`ffast-math` on GCC works by assuming you'll never get an inf, NAN, or a denorm number, which is ok up until it starts breaking `std::` functions that should always work cause they check bits)